gr-qtgui/lib/matrix_display.cc

```diff
@@ -0,0 +1,348 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2023 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#include "matrix_display.h"
+#include <QDebug>
+#include <vector>
+
+matrix_display::matrix_display(const std::string& name,
+                               unsigned int num_cols,
+                               unsigned int vlen,
+                               bool contour,
+                               const std::string& color_map,
+                               const std::string& interpolation,
+                               double x_start,
+                               double x_end,
+                               double y_start,
+                               double y_end,
+                               double z_max,
+                               double z_min,
+                               const std::string& x_axis_label,
+                               const std::string& y_axis_label,
+                               const std::string& z_axis_label,
+                               QWidget* parent)
+    : QWidget(parent),
+      d_vlen(vlen),
+      d_num_cols(num_cols),
+      d_x_start(x_start),
+      d_x_end(x_end),
+      d_y_start(y_start),
+      d_y_end(y_end),
+      d_z_max(z_max),
+      d_z_min(z_min)
+
+
+{
+    d_isclosed = false;
+    d_hLayout = nullptr;
+
+    d_menu = nullptr;
+    d_save_act = nullptr;
+    d_stop_act = nullptr;
+    d_contour_menu = nullptr;
+    d_interpolation_menu = nullptr;
```

Comment:
would put these constant initializations into the class declaration, i.e. in the .h file